### PR TITLE
Update uganda_releases

### LINKED ISSUE #594: Uganda: add financial year as date argument and update API domain
Some countries have financial year or fiscal year formats in a year range as strings, for example '2019-2021' in [Uganda](https://docs.google.com/spreadsheets/d/10tVioy-VOQa1FwWoRl5e1pMbGpiymA0iycNcoDFkvks/edit). This allows to implement the date spider arguments as strings.

Update the API domain to https://gpppapi.com

### PR DESCRIPTION
**Pull request by aguilerapy**
closes #594

### CODE REVIEW

File: kingfisher_scrapy/spiders/uganda_releases.py [hunk 1]
```diff
@@ -8,37 +8,55 @@ class Uganda(IndexSpider):
...
+        Download only data until this year (YYYY format). Defaults to the current year.
+        The year refers to the start of the fiscal year range, e.g. if until_date='2017' then fiscal_year='2017-2018'
     API documentation
       https://docs.google.com/spreadsheets/d/10tVioy-VOQa1FwWoRl5e1pMbGpiymA0iycNcoDFkvks/edit#gid=365266172
     """
     name = 'uganda_releases'
-    download_delay = 0.9
+    download_delay = 30  # to avoid API 429 error "too many request"
+    date_format = 'year'
+    custom_settings = {
```
[the remark this answers is not in the document]
Comment (aguilerapy) [reply kingfisher_scrapy/spiders/uganda_releases.py:24]: done

File: kingfisher_scrapy/spiders/uganda_releases.py [hunk 2]
```diff
@@ -8,37 +8,55 @@ class Uganda(IndexSpider):
     """
     Domain
       Government Procurement Portal (GPP) of Public Procurement and Disposal of Public Assets Authority (PPDA)
+    Spider arguments
+      from_date
+        Download only data from this year onward (YYYY format). Defaults to '2017'.
```
[the remark this answers is not in the document]
Comment (aguilerapy) [reply kingfisher_scrapy/spiders/uganda_releases.py:13]: done

File: kingfisher_scrapy/spiders/uganda_releases.py [hunk 3]
```diff
@@ -8,37 +8,55 @@ class Uganda(IndexSpider):
...
     def parse_data(self, response):
-        pattern = 'https://gpp.ppda.go.ug/adminapi/public/api/open-data/v1/releases/{}?fy={}&pde={}'
+        pattern = 'https://gpppapi.com/adminapi/public/api/open-data/v1/releases/{}?fy={}&pde={}'
 
         data = response.json()
         for pdes in data['data']['data']:
             for plans in pdes['procurement_plans']:
                 for tag in ('planning', 'tender', 'award', 'contract'):
+                    if self.from_date and self.until_date:
+                        start_year = int(plans['financial_year'].split("-")[0])
```
[the remark this answers is not in the document]
Comment (aguilerapy) [reply kingfisher_scrapy/spiders/uganda_releases.py:57]: done

File: kingfisher_scrapy/spiders/uganda_releases.py [hunk 4]
```diff
@@ -8,37 +8,57 @@ class Uganda(IndexSpider):
     """
     Domain
       Government Procurement Portal (GPP) of Public Procurement and Disposal of Public Assets Authority (PPDA)
-    API documentation
-      https://docs.google.com/spreadsheets/d/10tVioy-VOQa1FwWoRl5e1pMbGpiymA0iycNcoDFkvks/edit#gid=365266172
```
Comment (yolile) [inline kingfisher_scrapy/spiders/uganda_releases.py:12]: why are we removing this? I have access to the document, if the problem is that the domain described in each endpoint is no the correct one, I think that you can explain that here in a ["Caveats"](https://kingfisher-collect.readthedocs.io/en/latest/spiders.html#spider-metadata) section

File: kingfisher_scrapy/spiders/uganda_releases.py [hunk 5]
```diff
@@ -8,37 +8,61 @@ class Uganda(IndexSpider):
...
+        Download only data from this year onward (YYYY format).
+        If ``until_date`` is provided, defaults to '2017'.
+        The year refers to the start of the fiscal year range, e.g. if ``from_date`` = '2017' then the fiscal year is
+        '2017-2018'
+      until_date
+        Download only data until this year (YYYY format).
+        If ``from_date`` is provided, defaults to the current year.
+        The year refers to the start of the fiscal year range, e.g. if ``until_date`` = '2017' then the fiscal year is
+        '2017-2018'
+      API documentation
```
Comment (yolile) [inline kingfisher_scrapy/spiders/uganda_releases.py:24]: this one is indented incorrectly

**Review by yolile — APPROVED** [review]

**Pull request merged**